api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -417,71 +407,37 @@ private void handleWorkspaceLimit(long userId) {
     userRecentWorkspaceDao.deleteByUserIdAndWorkspaceIdIn(userId, idsToDelete);
   }
 
-  // this is necessary because the grant ownership call in create/clone
-  // may not have propagated. Adding a few retries drastically reduces
-  // the likely of failing due to slow propagation
-  private Retryer<ProjectBillingInfo> cloudBillingRetryer =
-      RetryerBuilder.<ProjectBillingInfo>newBuilder()
-          .retryIfException(
-              e ->
-                  e instanceof GoogleJsonResponseException
-                      && ((GoogleJsonResponseException) e).getStatusCode() == 403)
-          .withWaitStrategy(WaitStrategies.exponentialWait())
-          .withStopStrategy(StopStrategies.stopAfterDelay(60, TimeUnit.SECONDS))
-          .build();
-
   @Override
   public void updateWorkspaceBillingAccount(DbWorkspace workspace, String newBillingAccountName) {
     if (!workbenchConfigProvider.get().featureFlags.enableBillingUpgrade
         || newBillingAccountName.equals(workspace.getBillingAccountName())) {
       return;
     }
 
-    Cloudbilling cloudbilling;
     if (newBillingAccountName.equals(
         workbenchConfigProvider.get().billing.freeTierBillingAccountName())) {
-      cloudbilling = serviceAccountCloudbillingProvider.get();
+      fireCloudService.updateBillingAccountAsService(
+          workspace.getWorkspaceNamespace(), newBillingAccountName);
     } else {
-      cloudbilling = endUserCloudbillingProvider.get();
       try {
         Optional<Boolean> isOpenMaybe =
             Optional.ofNullable(
-                cloudbilling.billingAccounts().get(newBillingAccountName).execute().getOpen());
+                cloudBillingClient.getBillingAccount(newBillingAccountName).getOpen());
         boolean isOpen = isOpenMaybe.orElse(false);
         if (!isOpen) {
           throw new BadRequestException(
```

Comment:
I'm curious whether Firecloud just does this validation already - can we depend on that instead?

Reply:
I don't think FC does that, but I will double check with them. Ideally this should be a check happens in FC(not Google calls from AoU at all), if they do not have, we should ask for the support.  But I doubt the value they see having this check. If account closed, user will see less-friendly error while using Terra/RW. I think that's their take.